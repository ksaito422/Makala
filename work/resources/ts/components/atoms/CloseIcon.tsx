import React from 'react';
import { IconButton } from '@material-ui/core';
import { Close } from '@material-ui/icons';

type CloseIconProps = {
  onClick: (event: any) => void,
}

export const CloseIcon: React.FC<CloseIconProps> = ({
  onClick,
}) => {
  return (
    <IconButton
      component='span'
      onClick={onClick}
    >
      <Close />
    </IconButton>
  )
}