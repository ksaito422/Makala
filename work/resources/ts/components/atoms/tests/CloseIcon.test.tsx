import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { CloseIcon } from '../CloseIcon';

describe('CloseIcon', () => {
  it('スナップショットテスト', () => {
    const props = {
      onClick: jest.fn,
    };

    const { baseElement } = render(<CloseIcon {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
  it('clickイベントの確認', () => {
    const props = {
      onClick: jest.fn,
    };

    // onClickの監視
    // propsを受け取りrender テスト対象component生成
    const spyOnClick = jest.spyOn(props, 'onClick');
    const component = render(<CloseIcon {...props} />);
    // screen.debug();

    // <button>を探す
    // clickイベント実行
    // spyOnClickが実行されたか確認
    const button = component.getByRole('button');
    fireEvent.click(button);
    expect(spyOnClick).toHaveBeenCalled();
  });
});
