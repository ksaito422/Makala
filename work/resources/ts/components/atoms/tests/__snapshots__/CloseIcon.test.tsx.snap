// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseIcon スナップショットテスト 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;
